virttest/qemu_vm.py

```diff
@@ -1669,8 +1685,21 @@ def sort_key(dev):
                 LOG.warn(txt)
 
         smp_err = ""
+        SMP_PREFER_CORES_VERSION_SCOPE = '[6.2.0, )'
+        #  In the calculation of omitted sockets/cores/threads: we prefer
+        #  sockets over cores over threads before 6.2, while preferring
+        #  cores over sockets over threads since 6.2.
+        vcpu_prefer_sockets = params.get("vcpu_prefer_sockets")
+        if vcpu_prefer_sockets:
+            vcpu_prefer_sockets = params.get_boolean("vcpu_prefer_sockets")
+        else:
+            if utils_qemu.get_qemu_version(qemu_binary)[0] not in \
+                    VersionInterval(SMP_PREFER_CORES_VERSION_SCOPE):
+                # Prefer sockets over cores before 6.2
+                vcpu_prefer_sockets = True
+
```

Comment:
If `vcpu_prefer_sockets` does not match all judgment, then it equal to `params.get("vcpu_prefer_sockets")`, this means it's True, so L1735(if vcpu_prefer_sockets:) could not work normally.